### DISCUSSION TITLE
Decide on example data to include

### DISCUSSION THREAD

**Discussion by elray1** [hubverse-org/hubExamples#1] [General]
We would like to include enough data to be able to give meaningful examples across hubverse packages, but not so much that the package becomes unnecessarily bulky.

Here's an initial proposal for data that come from the example-complex-forecast-hub:

- 2 or 3 model_ids: Examples in hubVis and hubEnsembles benefit from having multiple models. I would vote for including three models, since examples of visualizations and ensembles with only two models might not be that interesting. But 2 might be sufficient.
- 2 locations: Functionality in hubVis benefits from having multiple locations. Maybe choose locations with different populations or signal/noise ratios? TX and MA?
- 2 reference_dates
- 2-4 horizons
- All output_types (to eventually include pmf, cdf, quantile, mean, median, sample; currently, there are no samples there)
- For quantile forecasts, 7 quantile levels sufficient for computing some scores and creating plots: 0.025, 0.1, 0.25, 0.5, 0.75, 0.9, 0.975

Would (subsets of) the product of these things be enough data to use as examples in hubverse packages?  Are there places where we need to add more?

What should this look like for examples from scenario modeling?

**Comment by nickreich** [comment]
I basically think that we should move ahead with the above scope for some example data from a forecast hub. It's a good place to start, and I think covers a lot of use cases. 

I will note that the example data in hubVis (based on contributions to [this spreadsheet](https://docs.google.com/spreadsheets/d/1sUn-oyyRCk9Pex_xawV_xg_RPy9QfZq8MOTlL7eXa74/edit#gid=1152681771) from @LucieContamin ) I think contains

 - 3 models
 - 1 location
 - 4 scenario ids
 - 1 origin date
 - 26 daily horizons
 - 1 target

Noting that @eahowerton had suggested a possible use case where a larger dataset would be useful, namely something like trying out a color palette option when coloring by model and you need more than 2 or 3 models to test it out.

**Reply by LucieContamin** [reply]
For HubVis, I created the example with the complex scenario hub example repo, I can provide the code if necessary!

**Comment by eahowerton** [comment]
I think this seems reasonable. The current `hubEnsembles` vignette always filters to a single location and origin date for example tables/plots. 

For horizons, it seems like our examples will likely fall into one of two cases: (1) we want to show a single forecast (i.e., 1 horizon), or (2) we want to show multiple horizons at the same time. If we are in case (1), we'll have to filter regardless of whether there are 2 or 4 horizons included. But, for case (2), I think there are benefits to the user of visualizing 4 weeks weeks ahead.

For the number of models, I agree having three models might be a sweet spot (especially if one of those models is a baseline model). I think it's useful to show variation in model predictions. I did raise the possibility of wanting more models for plotting (because we are often in situations with many more models than 3), but perhaps this is a case where we can use the full example-hub data. Not sure it's worth bogging down our small example files for something like this.

**Comment by elray1** [comment]
great, thanks, both!

noting that actually currently the example-complex-forecast-hub only has 3 models in it right now.  If it becomes a priority to add more at some point, we can do that.
